src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/AbstractRunCodeAnalysisCommand.cs

```diff
@@ -0,0 +1,203 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Packaging;
+using Microsoft.VisualStudio.ProjectSystem.Build;
+using Microsoft.VisualStudio.ProjectSystem.Input;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using IOleCommandTarget = Microsoft.VisualStudio.OLE.Interop.IOleCommandTarget;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Input.Commands
+{
+    internal abstract class AbstractRunCodeAnalysisCommand : AbstractSingleNodeProjectCommand, IVsUpdateSolutionEvents, IDisposable
+    {
+        private readonly IProjectThreadingService _threadingService;
+        private readonly IServiceProvider _serviceProvider;
+        private readonly RunCodeAnalysisBuildPropertyProvider _runCodeAnalysisBuildPropertyProvider;
+
+        private IVsSolutionBuildManager2 _buildManager;
+        private uint _solutionEventsCookie;
+        private IVsShell _vsShell;
+        private bool? _isCodeAnalysisPackageInstalled;
+        private bool _codeAnalysisPackageLoadAttempted;
+        private IOleCommandTarget _codeAnalysisCommandTarget;
+
+        protected AbstractRunCodeAnalysisCommand(
+            UnconfiguredProject unconfiguredProject,
+            IProjectThreadingService threadingService,
+            SVsServiceProvider serviceProvider,
```

Comment:
We normally do this as `[Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider`